h/static/scripts/app.js

```diff
@@ -82,12 +82,17 @@ function configureRoutes($routeProvider) {
   });
 }
 
+function isSidebar() {
+  var path = document.location.pathname;
+  return !path.startsWith('/stream') && !path.startsWith('/a/');
+}
+
 // @ngInject
 function setupFrameSync(frameSync) {
-  // Setup the connection to the frame hosting the sidebar app.
-  // This should only be done if this is the sidebar app, not the stream or
-  // standalone annotation pages.
-  return frameSync.connect();
+  // Setup connection to host frame
+  if (isSidebar()) {
```

Comment:
Agreed. I did some testing and it appears that the process of "discovering" connected frames takes 300-600ms, so we definitely want to make this call as early as possible and concurrently with fetching the session state via the '/app' call - which means before `WidgetController` gets loaded.

I'll have a look and see what we can do - I think it'll suffice to put it where we initialize the `$scope.isSidebar` property.